fossasia/visdom
# [feat] refactor data storage to allow saving on a per-window basis

## Description

- changed eager mode logic 
- changed data saving and loading logic
    - old logic in `SimpleJsonStorage`
    - per-window logic in `SimpleWindowJsonStorage`
    - using hard-coded MRO to switch between the logic for now.
- added a utility script for testing
- added auto directory creation
- added DEBUG utility printing

## Motivation and Context

#861 . changing towards a per-window caching scheme

## How Has This Been Tested?

- `rm -rf ./test_data/*`
- start server `python server.py -env_path test_data`
- `python test_simple.py` create plots an save
- closed server 
- reopen server
- check `env:test` is correctly loaded

## Screenshots (if appropriate):

```
tree test_data/
test_data/
├── test
│   ├── jsons
│   │   ├── 0.json
│   │   ├── 1.json
│   │   ├── 2.json
│   │   ├── 3.json
│   │   └── 4.json
│   └── reload.json
└── view

```

## Types of changes

## Checklist:
